### PR TITLE
Handle out of order sync calls 

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
@matriv please review

File: sql/src/main/java/io/crate/action/sql/SQLOperations.java [hunk 1]
```diff
@@ -143,8 +143,8 @@ public Session createSession(@Nullable String defaultSchema, Set<Option> options
         private Statement statement;
         private String query;
 
-        private Map<String, Portal> portals = new HashMap<>();
-        private String lastPortalName;
+        private final Map<String, Portal> portals = new HashMap<>();
+        private final Set<Portal> pendingExecutions = Collections.newSetFromMap(new IdentityHashMap<Portal, Boolean>());
```
Comment (matriv) [inline sql/src/main/java/io/crate/action/sql/SQLOperations.java]: Do we need LinkedHashSet to keep the order of portals inserted and execute the last one in sync?

Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/SQLOperations.java]: Currently there should be at most 1 entry so it doesn't really matter - I can also change it to a HashSet for now if you want

Comment (matriv) [inline sql/src/main/java/io/crate/action/sql/SQLOperations.java]: If we don't care about ordering no need. actually we would need LIFO (stack) to always pop the latest entry.

**Comment by matriv** [comment]
Igtm

**Pull request merged**